felix/calc/calc_graph_test.go

```diff
@@ -70,7 +67,20 @@ var _ = DescribeTable("Calculation graph pass-through tests",
 			},
 		})
 		eb.Flush()
-		Expect(reflect.ValueOf(messageReceived).Elem().Interface()).To(Equal(expUpdate))
+		equal := false
+		switch messageReceived.(type) {
+		case *proto.IPAMPoolUpdate:
+			equal = googleproto.Equal(messageReceived.(*proto.IPAMPoolUpdate), expUpdate.(*proto.IPAMPoolUpdate))
+		case *proto.HostMetadataUpdate:
+			equal = googleproto.Equal(messageReceived.(*proto.HostMetadataUpdate), expUpdate.(*proto.HostMetadataUpdate))
+		case *proto.GlobalBGPConfigUpdate:
+			equal = googleproto.Equal(messageReceived.(*proto.GlobalBGPConfigUpdate), expUpdate.(*proto.GlobalBGPConfigUpdate))
+		case *proto.WireguardEndpointUpdate:
+			equal = googleproto.Equal(messageReceived.(*proto.WireguardEndpointUpdate), expUpdate.(*proto.WireguardEndpointUpdate))
+		case *proto.ServiceUpdate:
+			equal = googleproto.Equal(messageReceived.(*proto.ServiceUpdate), expUpdate.(*proto.ServiceUpdate))
```

Comment:
Would it be more clear to duplicate `Expect(equal).To(BeTrue())` within each switch case? When the assertion fails, we immediately know which update message failed.

Reply:
updated.